### ISSUE TITLE
A few false positives

### ISSUE THREAD

**Issue by efx** [vastec/ember-unused-components#15]
Hi, thank you for building and share this script!

I wanted to note a few false positives I found running this on my app. These cases require advanced analysis and perhaps @lifeart's [ember-meta-explorer](https://github.com/lifeart/ember-meta-explorer) better suits finding those occurrences. [Dyfactor](https://github.com/dyfactor/dyfactor#why-dyfactor) could also apply here.

1. we invoke a component name as an argument to an action
```hbs
{{action "someAction" "my-component-form" bubbles=false}} 
```
2. we have a component only used in tests (very lame, I know 😄).
3. We have multiple `{{component (concat "thing-" someVariable)}}`

There are a couple others, mostly around more complex `{{component ` helper usage.
To me, a good solution would just be indicating possible false positives around any `{{component ` usage. Perhaps even more naive checks for the string of a component name used by helpers or passed as properties to a component.

**Comment by bryanhickerson** [comment]
Another false positive due to white space: `{{ close-button }}`

**Comment by tniezurawski** [comment]
@efx @bryanhickerson Thanks guys! I'll take a look into it 🙌 

@efx 
> These cases require advanced analysis and perhaps @lifeart's ember-meta-explorer better suits finding those occurrences. Dyfactor could also apply here.

Agree, `ember-meta-explorer` looks good. I took a path of simple (naive) regex patterns and didn't expect anything fancier would be needed. I'll see if I still can handle false positives you've reported with adjusting regexes of if I should change the strategy completely.

> 1. we invoke a component name as an argument to an action
> `{{action "someAction" "my-component-form" bubbles=false}}`

Tricky! 😄 

> 2. we have a component only used in tests (very lame, I know 😄).

Hmm 🤔, could you tell me a bit more? Now I'm curious why you had to do it.

> 3. We have multiple {{component (concat "thing-" someVariable)}}

That's out of the scope unfortunately. The variable can come from the server response, like:

```
// Campaign types: email, social-media, sms (etc...)
// Response from the server: campaign.type: "email"

{{component (concat "campaign-wizard-" campaign.type)}}
```
... and there is no way I can analyze that.

But I like the idea of a "fuzzy" check and indicating that maybe a component is used in this line. But due to dynamic name resolution, we can't be sure.

**Comment by efx** [comment]
Certainly! Most of the ones I saw are caveats and shouldn't necessarily be fixed in this tool; I think documenting them just serves future users who may expect otherwise.

Per point 2, we have a table component that takes a property of the component name to render each row. Within that table component we invoke the provided row component like so:
```hbs
  {{component rowComponent item=item}}
```
So another case of dynamic `{{component` usage. 😆. I honestly need to remove that pattern.

**Comment by tniezurawski** [comment]
I'm not yet sure if I can handle all the cases but I'm leaving that issue open to at least fix the thing @bryanhickerson was writing about:
> Another false positive due to white space: {{ close-button }}

**Comment by miguelcobain** [comment]
Another false positive are components in comments:
```hbs
{{!--
  {{my-button}}
--}}
```
